javascript.go

```diff
@@ -0,0 +1,174 @@
+package flixkit
+
+import (
+	"bytes"
+	"embed"
+	"strings"
+	"text/template"
+	"unicode"
+)
+
+type SimpleParameter struct {
+	Name string
+	JsType string
+    Description string
+    FclType string
+    CadType string
+}
+
+type TemplateData struct {
+    Version     string
+    Parameters []SimpleParameter
+    Title       string
+    Description string
+    Location    string
+    IsScript    bool
+    IsLocalTemplate bool
+}
+
+//go:embed templates/*.tmpl
+var templateFiles embed.FS
```

Comment:
In general i advice putting embeds in the main go file, it is so much easier to have an overview of where they are done that way. That also makes it easier to test templates that are wrongly formted since you can mock fs with MapFs from https://pkg.go.dev/testing/fstest

Reply:
Not sure this templates will cary over to flow-cli if we import it there. I have not tested go embeds in dependencies.

Reply:
seems to be working when I reference flixkit-go from flow-cli locally. Is this a good test?